File: Command/Java/src/command/RemoteControl.java
package command;

import java.util.Arrays;

public class RemoteControl {
    private Command[] onCommands;
    private Command[] offCommands;
    public RemoteControl(){
        onCommands = new Command[4];
        offCommands = new Command[4];
        NoCommand noCommand = new NoCommand();
        for(int i=0;i<onCommands.length;i++){
            onCommands[i] = noCommand;
            offCommands[i] = noCommand;
        }
    }
    public void addCommand(int slot , Command onCommand , Command offCommand){
        onCommands[slot] = onCommand;
        offCommands[slot] = offCommand;

    }
    public void onButtonPressed(int slot){
        onCommands[slot].execute();
    }
    public void offButtonPressed(int slot){
        onCommands[slot].undo();
    }

    @Override
    public String toString() {
        StringBuffer buffer = new StringBuffer();
        for(int i=0 ; i<onCommands.length ; i++){
            buffer.append(String.format("[slot %d] %s \t %s%n",i,onCommands[i].getClass().getSimpleName(),offCommands[i].getClass().getSimpleName()));

        }
        return buffer.toString();
    }
}

File: Command/Python/RemoteControl.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 18 22:15:05 2022

@author: Mahmoud Saeed
"""
import Command as command
import NoCommand as noCommand

class RemoteControl():
    onCommand = []
    offCommand = []
    
    def __init__(self):
        no_command = noCommand.NoCommand()
        for i in range(4):
            self.onCommand.append(no_command)
            self.offCommand.append(no_command)
            
        
    def addCommand(self , i , on_command , off_command):
              self.onCommand.insert(i ,on_command )  
              self.offCommand.insert(i ,off_command )
            
    def onButtonPressed(self , i):
         self.onCommand[i].excute()
     
    def offButtonPressed(self , i):
         self.onCommand[i].undo()
    

         